manifests/cert.pp

```diff
@@ -2,76 +2,149 @@
 
 define sslcert::cert (
   $certname,
-  $ensure   = 'present',
-  $csr      = '',
-  $certpath = $sslcert::certpath,
-  $chain    = '',
-  $owner    = 'root',
-  $group    = 'root',
-  $mode     = '0644',
-  $cert     = '',
-  $keypath  = $sslcert::keypath,
-  $keyname  = undef,
-  $keyowner = undef,
-  $keygroup = undef,
-  $keymode  = undef,
-  $key      = '',
+  $ensure              = 'present',
+  $csr                 = '', # not used, don't ask
+  $cert                = '', # actual cert
+  $certpath            = $::sslcert::certpath,# backward compatiblity (directory)
+  $certfile            = '', # complete path of file (dir+file)
+  $certowner           = undef, # default same as owner
+  $certgroup           = undef, # default same as group
+  $certmode            = undef, # default same as mode
+  $certtype            = '', # needed for digicert request
+  $chain               = '', # actual chain
+  $chainpath           = $::sslcert::certpath, #backward compatiblity (directory)
+  $chainfile           = '', # complete path of file (dir+file)
+  $chainowner          = undef, # default same as owner
+  $chaingroup          = undef, # default same as group
+  $chainmode           = undef, # default same as mode
+  $chainenabled        = $::sslcert::chainenabled,
+  $common_name         = '', # ex. something.com
+  $san_names           = '', # comma seperated list, common_name is required for this
+  $ssl_service_name    = '', # ex. apache
+  $ssl_service_command = '', # ex. restarted
+  $owner               = 'root', # default value for key,chain,cert
+  $group               = 'root', # default value for key,chain,cert
+  $mode                = '0644', # default value for chain,cert
+  $keypath             = $::sslcert::keypath, # backward compatiblity (directory)
+  $keyfile             = '', # complete path of file (dir+file)
+  $keyname             = undef, # default certname
+  $keyowner            = undef, # default same as owner
+  $keygroup            = undef, # default same as group
+  $keymode             = undef, # default value 0600
+  $key                 = '', # actual key
+  $renew_key           = '', # force key renewal with cert request
```

Comment:
Should be have a default set to true? And only accept true/false? (Which can be check with is_bool from stdlib)